sfdx-source/apex-common/test/classes/fflib_SObjectUnitOfWorkTest.cls

```diff
@@ -848,6 +905,43 @@ private with sharing class fflib_SObjectUnitOfWorkTest
         }
     }
 
+    private class MockDMLWithInsertIds implements fflib_SObjectUnitOfWork.IDML
+    {
+        public List<SObject> recordsForInsert = new List<SObject>();
+        public List<SObject> recordsForUpdate = new List<SObject>();
+        public List<SObject> recordsForDelete = new List<SObject>();
+        public List<SObject> recordsForRecycleBin = new List<SObject>();
+        public List<SObject> recordsForEventPublish = new List<SObject>();
+
+        public void dmlInsert(List<SObject> objList)
+        {
+            for (SObject obj : objList) {
+                obj.Id = fflib_IDGenerator.generate(obj.getSObjectType());
```

Comment:
@ClayChipps and @jprichter -- Please add a null check on `obj` before trying to generate the mock Id.

Reply:
done